plancheck/expect_resource_action.go

```diff
@@ -0,0 +1,85 @@
+package plancheck
+
+import (
+	"context"
+	"fmt"
+)
+
+var _ PlanCheck = expectResourceAction{}
+
+type expectResourceAction struct {
+	resourceAddress string
+	actionType      ResourceActionType
+}
+
+// CheckPlan implements the plan check logic.
+func (e expectResourceAction) CheckPlan(ctx context.Context, req CheckPlanRequest, resp *CheckPlanResponse) {
+	foundResource := false
+
+	for _, rc := range req.Plan.ResourceChanges {
+		if e.resourceAddress == rc.Address {
```

Comment:
Super nit: This logic can be reversed to prevent the nesting below

```go
if e.resourceAddress != rc.Address {
  continue
}

// nested logic :)
```

Reply:
inverted! ✈️ ⚡